rmf_traffic/test/unit/agv/test_Planner.cpp

```diff
@@ -458,6 +458,159 @@ SCENARIO("Test Options", "[options]")
+    add_bidir_lane(4, 5);
+    add_bidir_lane(6, 10);
+    add_bidir_lane(7, 8);
+    add_bidir_lane(9, 10);
+    add_bidir_lane(10, 11);
+    add_bidir_lane(5, 9);
+    add_bidir_lane(11, 12);
+
+    const rmf_traffic::Profile profile = create_test_profile(UnitCircle);
+    const rmf_traffic::agv::VehicleTraits traits(
+      {0.7, 0.3}, {1.0, 0.45}, profile);
+    auto interrupt_flag = std::make_shared<bool>(false);
+
+    WHEN("Goal from 12->1")
+    {
+      const rmf_traffic::Time time = std::chrono::steady_clock::now();
+      const auto start = rmf_traffic::agv::Planner::Start{time, 12, 0.0};
+      const auto goal = rmf_traffic::agv::Planner::Goal{1};
+
+      THEN("No cost limit succeeds")
+      {
+        auto options = rmf_traffic::agv::Planner::Options{nullptr};
+        rmf_traffic::agv::Planner planner{
+          rmf_traffic::agv::Planner::Configuration{graph, traits},
+          options
+        };
+        auto plan = planner.plan(start, goal);
+        REQUIRE(plan);
+        REQUIRE(plan.cost_estimate());
+      }
+
+      THEN("Cost limit of zero fails")
+      {
+        auto options = rmf_traffic::agv::Planner::Options{
+          nullptr,
+          rmf_traffic::agv::Planner::Options::DefaultMinHoldingTime,
+          interrupt_flag,
+          0 // Maximum cost estimate must be 0
+        };
+        rmf_traffic::agv::Planner planner{
+          rmf_traffic::agv::Planner::Configuration{graph, traits},
+          options
+        };
+        auto plan = planner.plan(start, goal);
+        CHECK(!plan);
+      }
+    }
+
+    WHEN("Goal from 12->1 and known required cost limit")
+    {
+      const rmf_traffic::Time time = std::chrono::steady_clock::now();
+      const auto start = rmf_traffic::agv::Planner::Start{time, 12, 0.0};
+      const auto goal = rmf_traffic::agv::Planner::Goal{1};
+
+      auto options = rmf_traffic::agv::Planner::Options{nullptr};
+      rmf_traffic::agv::Planner planner{
+        rmf_traffic::agv::Planner::Configuration{graph, traits},
+        options
+      };
+      auto plan = planner.plan(start, goal);
+      REQUIRE(plan);
+      REQUIRE(plan.cost_estimate());
+      auto required_cost = plan.cost_estimate().value();
+
+      THEN("Cost limit half actual required limit fails")
+      {
+
+        options.maximum_cost_estimate(required_cost / 2);
+        planner.set_default_options(options);
+        plan = planner.plan(start, goal);
+        CHECK(!plan);
+      }
+
+      THEN("Cost limit just below actual required limit fails")
+      {
+
+        options.maximum_cost_estimate(required_cost - 1e-6);
+        planner.set_default_options(options);
+        plan = planner.plan(start, goal);
+        CHECK(!plan);
+      }
+
+      THEN("Cost limit just above actual required limit succeeds")
+      {
+
+        options.maximum_cost_estimate(required_cost + 1e-6);
+        planner.set_default_options(options);
+        plan = planner.plan(start, goal);
+        CHECK(!plan);
```

Comment:
```suggestion
        CHECK(plan);
```

Looks like this expectation got accidentally flipped.